pkg/report/pb/data-feeds/on-chain/registry/decode.go

```diff
@@ -33,42 +29,18 @@ func DecodeAsFeedUpdated(m *wt_msg.WriteConfirmed) ([]*FeedUpdated, error) {
 
 	// Iterate over the underlying Mercury reports
 	for _, rf := range *reports {
-		// Decode the common Mercury report and get report type
-		rmCommon, err := mercury_vX.Decode(rf.Data)
-		if err != nil {
-			return nil, fmt.Errorf("failed to decode Mercury report: %w", err)
-		}
-
-		// Parse the report type from the common header
-		t := mercury_vX.GetReportType(rmCommon.FeedID)
 		feedID := datafeeds.FeedID(rf.FeedID)
 
-		switch t {
-		case uint16(3):
-			rm, err := mercury_v3.Decode(rf.Data)
-			if err != nil {
-				return nil, fmt.Errorf("failed to decode Mercury v%d report: %w", t, err)
-			}
-			// For Mercury v3, include TxSender and TxReceiver
-			msgs = append(msgs, newFeedUpdated(m, feedID, rm.ObservationsTimestamp, rm.BenchmarkPrice, rf.Data, true))
-		case uint16(4):
-			rm, err := mercury_v4.Decode(rf.Data)
-			if err != nil {
-				return nil, fmt.Errorf("failed to decode Mercury v%d report: %w", t, err)
-			}
-			// For Mercury v4, skip TxSender and TxReceiver (if not applicable)
-			msgs = append(msgs, newFeedUpdated(m, feedID, rm.ObservationsTimestamp, rm.BenchmarkPrice, rf.Data, false))
-		default:
-			return nil, fmt.Errorf("unsupported Mercury report type: %d", t)
-		}
+		// TODO: unsure if r.Data is correct for Report
+		msgs = append(msgs, NewFeedUpdated(m, feedID, rf.Timestamp, rf.Price, r.Data, false))
```

Comment:
Nope. It should be `rf.Data` (report feed data).

`r.Data` is platform report data -> data which contains an encoded list of feed updates data

Reply:
But there is no rf.Data for the EVM type, it follows this schema https://github.com/smartcontractkit/chainlink-evm/blob/b18ae9cad0b175959fe5fdc2f6c849ff22e789a6/pkg/report/datafeeds/types.go#L36-L38

Reply:
Ok, makes sense.

So for EVM we get a decoded report (part of it) and on Aptos we get a raw report that we decode ourselves (a Mercury report).

I'd say the correct way to express this is that we just didn't get a DF underlying raw report, so let's just attach an empty byte array there.